pynamodb/attributes.py

```diff
@@ -443,8 +442,10 @@ def _update_attribute_types(cls, attribute_values: Dict[str, Dict[str, Any]]):
     def _coerce_attribute_type(attr_type: str, attribute_value: Dict[str, Any]):
         # coerce attribute types to disambiguate json string and array types
         if attr_type == BINARY and STRING in attribute_value:
-            attribute_value[BINARY] = attribute_value.pop(STRING)
-        if attr_type in {BINARY_SET, NUMBER_SET, STRING_SET} and LIST in attribute_value:
+            attribute_value[BINARY] = base64.b64decode(attribute_value.pop(STRING))
+        elif attr_type == BINARY_SET and LIST in attribute_value:
+            attribute_value[BINARY_SET] = [base64.b64decode(v[STRING]) for v in attribute_value.pop(LIST)]
```

Comment:
**Why do we base64-decode here when we didn't use to?** Since `BinaryAttribute(legacy_encoding=False).deserialize` does not base64-decode (assuming `_convert_binary` in connection.py already did the job), we have to do it here for the sake of `Model.from_dict`.